src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -344,26 +344,21 @@ private BoundExpression CreateTupleConversion(CSharpSyntaxNode syntax, BoundTupl
 
             NamedTypeSymbol targetType = (NamedTypeSymbol)destinationWithoutNullable;
 
-            NamedTypeSymbol targetUnderlyingType;
             if (targetType.IsTupleType)
             {
                 var destTupleType = (TupleTypeSymbol)targetType;
-                targetUnderlyingType = destTupleType.UnderlyingTupleType;
                 // do not lose the original element names in the literal if different from names in the target
+
+                // PROTOTYPE(tuples): Come back to this, what about locations?
                 targetType = destTupleType.WithElementNames(sourceTuple.ArgumentNamesOpt);
```

Comment:
I agree, it doesn't. This is on my list to look at once symbol model is in the state we would like it to be. This activity is out of scope of this PR.